### ISSUE TITLE
weird behaviour, header not always interpreted correctly

### ISSUE THREAD

**Issue by kampfgnu** [xing/XNGMarkdownParser#10]
it is weird and i can't find the reason.
here are some non-working examples:

* if the word has no "\n" at the end, it does not work in all cases

does not work:
```XNGMarkdownParser *p = [[XNGMarkdownParser alloc] init];```
``` NSAttributedString *s = [p attributedStringFromMarkdownString:@"# foo"];```

works:
```XNGMarkdownParser *p = [[XNGMarkdownParser alloc] init];```
```NSAttributedString *s = [p attributedStringFromMarkdownString:@"# foo\n"];```

* if the word contains an "L" (upper and lower case)

does not work:
```XNGMarkdownParser *p = [[XNGMarkdownParser alloc] init];```
```NSAttributedString *s = [p attributedStringFromMarkdownString:@"# L\n"];```

works:
```XNGMarkdownParser *p = [[XNGMarkdownParser alloc] init];```
```NSAttributedString *s = [p attributedStringFromMarkdownString:@"# M\n"];```

(tested on xcode 6.2, iOS8)

**Comment by pietbrauer** [comment]
I think it may be related to https://github.com/xing/XNGMarkdownParser/issues/6 ?

**Comment by gskbyte** [comment]
Exactly, this is happening because the header requires a newline at the end:

```
header1        ^#{1,6}[^#{nl}]*$
header2        .+\n[=-]+\n
```

We will need to have a look at the grammar

**Comment by pietbrauer** [comment]
Fixed with #11
